src/WrappedMToken.sol

```diff
@@ -81,10 +81,14 @@ contract WrappedMToken is IWrappedMToken, Migratable, ERC20Extended {
 
     /// @inheritdoc IWrappedMToken
     function wrap(address recipient_, uint256 amount_) external {
+        uint256 startingBalance_ = IMTokenLike(mToken).balanceOf(address(this));
+
         // NOTE: The behavior of `IMTokenLike.transferFrom` is known, so its return can be ignored.
         IMTokenLike(mToken).transferFrom(msg.sender, address(this), amount_);
 
-        _mint(recipient_, UIntMath.safe240(amount_));
+        uint256 endingBalance_ = IMTokenLike(mToken).balanceOf(address(this));
+
+        _mint(recipient_, UIntMath.safe240(endingBalance_ - startingBalance_));
```

Comment:
Not sure about this approach.
Wouldn't you be capturing the M that has accrued in the wrapper?

Since `transferFrom` will update the index:
https://github.com/MZero-Labs/protocol/blob/b1c6e624ed09a9e28f4ae45cd87fda610fafe446/src/MToken.sol?plain=1#L402

The diff in balance will be the amount of M deposited plus any M earned since the last `updateIndex()` call.

Reply:
> Wouldn't you be capturing the M that has accrued in the wrapper?

No, because the M balance of any earning account does not change in a block other than for receiving or sending M. Same for the index. `updateIndex()` does not change the index, it simply locks in the current index as per the previous rate and updates the rate to use going forward.